### PR TITLE
Added Permissions

### PR DESCRIPTION
**Pull request by amoghparab247**

### CODE REVIEW

**Comment by vixrant** [comment]
> @vixrant suggest the changes in test.py

I'll add the tests that I've been working on after your PR gets merged.

File: canteenWeb/tests.py [hunk 1]
```diff
@@ -63,3 +72,134 @@ def test_order_reject(self):
...
+        cls.client = APIClient()
+        cls.client.force_authenticate(user=cls.user)
+
+    def setUp(self):
+        # Create an menu before all tests.
+        self.sandwich = MenuItem.objects.create(
+            name="Sandwich",
+            price=random.randint(0, 100),
+            is_available=True,
+            options="JAIN",
```
Comment (vixrant) [inline canteenWeb/tests.py]: You forgot a field which is required.

Before creating sandwich, create

```python

self.indian = Category.objects.create(
    name="Indian"
)

self.sandwich = MenuItem.object.create(
    ... # Whatever you have
    category=self.indian
)

```

File: canteenWeb/urls.py [hunk 2]
```diff
@@ -23,4 +23,9 @@
 router.register("bill", views.BillViewSet)
 router.register("category", views.CategoryViewSet)
 
-urlpatterns = [path("", include(router.urls))]
+urlpatterns = [
+    path("", include(router.urls)),
+    path("signup/", views.SignUp.as_view(), name="signup"),
```
Comment (vixrant) [inline canteenWeb/urls.py]: Nice catch...

File: canteenWeb/views.py [hunk 3]
```diff
@@ -19,12 +19,22 @@
 )
 
 
+class IsAdminUserOrReadOnly(permissions.BasePermission):
+    def has_permission(self, request, view):
+        return (
+            request.method in permissions.SAFE_METHODS
```
Comment (preet4737) [inline canteenWeb/views.py]: Just a suggestion - Why not make another file just for permissions?

File: canteenWeb/views.py [hunk 4]
```diff
@@ -116,12 +139,14 @@ def post(self, request):
                 password=serializer.data.get("password"),
             )
             login(request, user)
-            return HttpResponseRedirect(redirect_to="/menu_item/")
+            return HttpResponseRedirect(
```
Comment (vixrant) [inline canteenWeb/views.py]: This isn't required really cause I think the default status is 200 OK...

File: canteenWeb/tests.py (hunk 1 above)
Comment (amoghparab247) [reply canteenWeb/tests.py]: Added it.

File: canteenWeb/tests.py [hunk 5]
```diff
@@ -16,7 +23,9 @@ def setUpClass(cls):
         cls.user = User.objects.create(
             username="Student", email="sixty@nine.com", password="super-secret-meow"
```
Comment (preet4737) [inline canteenWeb/tests.py]: remove the email parameter

**Comment by aayushkothari11** [comment]
@amoghparab247 status regarding this PR? Are you stuck somewhere?

**Pull request merged**